woodwork/tests/accessor/test_table_accessor.py

```diff
@@ -844,3 +844,96 @@ def test_is_valid_schema(sample_df):
 
     different_dtype_df = schema_df.astype({'id': 'Int64'}, copy=True)
     assert not _is_valid_schema(different_dtype_df, schema)
+
+
+def test_dataframe_methods_on_accessor(sample_df):
+    xfail_dask_and_koalas(sample_df)
+
+    schema_df = sample_df.copy()
+    schema_df.ww.init(name='test_schema')
+
+    copied_df = schema_df.ww.copy()
+
+    assert schema_df is not copied_df
+    assert isinstance(copied_df.ww.schema, Schema)
+    assert copied_df.ww.schema == schema_df.ww.schema
+
+    pd.testing.assert_frame_equal(to_pandas(schema_df), to_pandas(copied_df))
+
+    error = 'Woodwork typing information is not valid for this DataFrame.'
+    with pytest.raises(ValueError, match=error):
+        schema_df.ww.astype({'id': 'string'})
+
+
+def test_dataframe_methods_on_accessor_inplace(sample_df):
+    xfail_dask_and_koalas(sample_df)
+
+    schema_df = sample_df.copy()
+    schema_df.ww.init(name='test_schema')
+
+    df_pre_sort = schema_df.copy()
+
+    schema_df.ww.sort_values(['full_name'], inplace=True)
+    assert schema_df.ww.name == 'test_schema'
+
+    pd.testing.assert_frame_equal(to_pandas(schema_df), to_pandas(df_pre_sort.sort_values(['full_name'])))
+
+    error = 'Woodwork typing information is not valid for this DataFrame.'
+    with pytest.raises(ValueError, match=error):
+        schema_df.ww.rename({'id': 'new_name'}, inplace=True, axis=1)
+    assert 'new_name' in schema_df.columns
```

Comment:
Worth noting that even when the schema gets invalidated and we throw an error, it's happening after we've made the call, so in the case of inplace operations if you were to catch the error, you'd still have the action performed. I'm not sure how we'd avoid this, but maybe it's not a problem?

(can also see this in the call of `df.ww.pop` in `test_dataframe_methods_on_accessor_returning_series`)

Reply:
Would you be ok with setting `df.ww.schema` to None any time the schema is invalidated? Is there a reason to keep it present at all (maybe to allow updating it so that it's valid again)?

When it comes to checking that the schema is valid before being able to access it, we can add the check at the top level of the `__getattr__`. That means that we'll need to explicitly check the validity of the schema in any Accessor methods. I do think it probably makes sense as a separate error from the `Woodwork not initialized for this DataFrame` one, so maybe that's reason enough to not just set the schema to None when it's invalidated. 

Would it be worth raising a warning when the schema gets invalidated? Trying to think about a situation in which you might perform a sequence of operations on a DataFrame and it'd be useful to know which one was causing the schema to be invalidated.

Reply:
So that's a good question. Maybe it would be best for now just to set the schema back to None if it is determined to be invalid. If we could provide a meaningful warning (even if it is somewhat generic) at that time we reset, that might be sufficient for now. We could always add more later if it becomes a bigger issue.

The more I think about it, setting the schema back to None seems cleaner than sprinkling a bunch of schema validation checks throughout the code.